projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/Sentence.kt

```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.digital.hmpps.integrations.delius.sentence.entity
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import uk.gov.justice.digital.hmpps.integrations.delius.overview.entity.Event
+
+interface EventSentenceRepository : JpaRepository<Event, Long> {
+
+    @Query(
+        "SELECT e FROM Event e " +
+            "LEFT JOIN FETCH e.disposal d " +
+            "LEFT JOIN FETCH d.type t  " +
+            "LEFT JOIN FETCH e.mainOffence m " +
+            "LEFT JOIN FETCH e.additionalOffences ao " +
+            "LEFT JOIN FETCH m.offence mo " +
+            "LEFT JOIN FETCH ao.offence aoo " +
+            "WHERE e.personId = :personId"
+    )
+    fun findPersonById(personId: Long): List<Event>
```

Comment:
If this is a new repository only used by getSentenceDetails, we can just do a join on the offender table (or a new stripped down sentence.entity.Person entity. This will mean we dont need to use the overview personRespository later

Reply:
done